browser/main/modals/PreferencesModal/BackupTab.js

```diff
@@ -0,0 +1,248 @@
+  ? {
+    title: i18n.__('Select Directory'),
+    defaultPath,
+    properties: ['openDirectory', 'createDirectory']
+  }
+  : {
+    title: i18n.__('Select File'), defaultPath
+  }
+  return new Promise((resolve, reject) => {
+    dialog.showOpenDialog(browseProperties, function (targetPaths) {
+      if (targetPaths == null) return resolve('')
+      resolve(targetPaths[0])
+    })
+  })
+}
+
+class BackupTab extends React.Component {
+  constructor (props) {
+    super(props)
+
+    this.state = {
+      config: props.config,
+      page: 'HOME',
+      newExport: {
+        path: ''
+      },
+      newImport: {
+        path: ''
+      }
+    }
+  }
+
+  componentDidMount () {
+    this.handleSettingDone = () => {
+      this.setState({BackupAlert: {
+        type: 'success',
+        message: i18n.__('Successfully applied!')
+      }})
+    }
+    this.handleSettingError = (err) => {
+      this.setState({BackupAlert: {
+        type: 'error',
+        message: err.message != null ? err.message : i18n.__('An error occurred!')
+      }})
+    }
+    ipc.addListener('APP_SETTING_DONE', this.handleSettingDone)
+    ipc.addListener('APP_SETTING_ERROR', this.handleSettingError)
+  }
+
+  componentWillUnmount () {
+    ipc.removeListener('APP_SETTING_DONE', this.handleSettingDone)
+    ipc.removeListener('APP_SETTING_ERROR', this.handleSettingError)
+  }
+
+  handleSaveImport (e) {
+    const newConfig = {
+      hotkey: this.state.config.hotkey,
+      ui: this.state.config.ui,
+      editor: this.state.config.editor,
+      preview: this.state.config.preview,
+      blog: this.state.config.blog
+    }
+
+    ConfigManager.set(newConfig)
+
+    store.dispatch({
+      type: 'SET_UI',
+      config: newConfig
+    })
+    this.clearMessage()
+    this.props.haveToSave()
+  }
+
+  handleImportPathBrowseButtonClick (e) {
+    browseFolder(false)
+      .then((targetPath) => {
+        if (targetPath.length > 0) {
+          const { newImport } = this.state
+          newImport.path = targetPath
+          this.setState({
+            newImport
+          })
+        }
+      })
+      .catch((err) => {
+        console.error('BrowseFAILED')
+        console.error(err)
+      })
+  }
+
+  handleExportPathBrowseButtonClick (e) {
+    browseFolder(true)
+      .then((targetPath) => {
+        if (targetPath.length > 0) {
+          const { newExport } = this.state
+          newExport.path = targetPath
+          this.setState({
+            newExport
+          })
+        }
+      })
+      .catch((err) => {
+        console.error('BrowseFAILED')
+        console.error(err)
+      })
+  }
+
+  importConfig (e) {
+    if (this.state.newImport.path === '') {
+      alert(i18n.__('No file has been selected.'))
+      return false
+    }
+
+    try {
+      const externalConfig = JSON.parse(fs.readFileSync(this.state.newImport.path, 'utf-8'))
+      const { config } = this.state
+      config.hotkey = externalConfig.hotkey
```

Comment:
Can you add a inner try catch here to validate the config too? Because otherwise the error message will be `File is not valid JSON.` which will confuse the user.